### PR TITLE
Fill Mask Pipeline to HuggingFace Model

### PR DESCRIPTION
**Pull request by dhuvik**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/hf_models.py [hunk 1]
```diff
@@ -502,3 +502,82 @@ def _predict(self, generator: Iterable[Tuple[Any, Any, Any]],
...
+        -------
+        Union[List[Dict], List[List[Dict]]]
+            A list or a list of list of dictionaries with the following keys:
+            - sequence (str): The corresponding input with the mask token prediction.
+            - score (float): The corresponding probability.
+            - token (int): The predicted token id (to replace the masked one).
+            - token_str (str): The predicted token (to replace the masked one).
+
+
+        DEV NOTES:
```
Comment (rbharath) [inline deepchem/models/torch_models/hf_models.py:529]: Replace with numpydoc style Notes section

File: deepchem/models/torch_models/hf_models.py [hunk 2]
```diff
@@ -502,3 +502,82 @@ def _predict(self, generator: Iterable[Tuple[Any, Any, Any]],
...
+                generator: Iterable[Tuple[Any, Any, Any]],
+                 transformers: List[Transformer], uncertainty: bool,
+                 other_output_types: Optional[OneOrMany[str]]
+        '''
+
+        and then re-expand them back out again.
+
+        """
+        self._ensure_built()
+        self.model.eval()
```
Comment (rbharath) [inline deepchem/models/torch_models/hf_models.py:549]: Add some intermediate comments; the code is long enough it's hard to follow

File: deepchem/models/torch_models/tests/test_hf_models.py [hunk 3]
```diff
@@ -201,3 +201,45 @@ def test_load_from_hf_checkpoint():
 
     # keys should not match
     assert all(not_matches)
+
+
+@pytest.mark.torch
+def test_fill_mask(tmpdir, hf_tokenizer):
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_hf_models.py:207]: Maybe 2 separate unit tests

**Review by rbharath — COMMENTED** [review]
Few changes; should be good once done

File: deepchem/models/torch_models/hf_models.py (hunk 1 above)
Comment (dhuvik) [reply deepchem/models/torch_models/hf_models.py]: Updated and fixed.

File: deepchem/models/torch_models/hf_models.py (hunk 2 above)
Comment (dhuvik) [reply deepchem/models/torch_models/hf_models.py:535]: Comments interleaved.

File: deepchem/models/torch_models/tests/test_hf_models.py (hunk 3 above)
Comment (dhuvik) [reply deepchem/models/torch_models/tests/test_hf_models.py]: Tests have been successfully split

**Review by rbharath — COMMENTED** [review]
LGTM once yapf fix

**Review by rbharath — APPROVED** [review]
LGTM
